setjmp.c

```diff
@@ -18,12 +19,12 @@ int main()
 	// buffer[14..31] = ???
 	for (uint32_t idx; idx < (length / 16); idx++)
 	{
-		if (cheri_is_valid(((void **)buffer)[idx]))
+		if (cheri_gettag(((void **)buffer)[idx]))
 		{
 			void *csp = cheri_csp_get();
 			uint64_t base = cheri_base_get(csp);
 			uint64_t length = cheri_length_get(csp);
-			uint64_t address = cheri_address_get(((void **)buffer)[idx]);
+			uint64_t address = cheri_getaddress(((void **)buffer)[idx]);
 
 			if (address >= base && address <= (base + length))
```

Comment:
Only just noticed this even though it's not a change:  The upper bound is [not inclusive](https://github.com/CTSRD-CHERI/newlib/blob/d92070d2367ad72dcbf161825abdc64a7a1ec4f0/newlib/libc/include/cheric.h#L90) so you might want to tweak it to `address < (base + length)`

Reply:
Fixed.